### DISCUSSION TITLE
Verification of iso changes on existing metalk8s deployment

### DISCUSSION THREAD

**Discussion by Abhishek627** [scality/metalk8s#2970] [Q&A] [answered]
If I want to validate a new iso generated, can I just mount it and redeploy on an existing setup ? If yes, does any existing file/folder needs to be purged before this redeployment for a fair test.

**Comment by gdemonet** [comment, marked as answer]
Hello @Abhishek627! Yes, you can replace an existing ISO with a new one on an existing deployment. We have a minimal procedure for doing so in our developer documentation: see it [here](https://metal-k8s.readthedocs.io/en/2.6.0/developer/deploy/upgrade.html).
Hope this helps!

**Comment by Abhishek627** [comment]
Hi ! There is 1 more issue I've seen popping up quite frequently, if you have some pointers for this, this'd be very helpful.
```
  Name: Ensure repositories container is up - Function: module.wait - Result: Clean Started: - 12:26:55.376192 Duration: 0.653 ms
----------
          ID: Wait for Repositories container to answer
    Function: http.wait_for_successful_query
        Name: http://10.105.58.111:8080/metalk8s-2.6.0-dev/
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1981, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1977, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/http.py", line 213, in wait_for_successful_query
                  raise caught_exception  # pylint: disable=E0702
              CommandExecutionError: [Errno 111] Connection refused
     Started: 12:26:55.377715
    Duration: 300001.418 ms
```
This ip is correct and normal curl doesn't work too.

**Reply by NicolasT** [reply]
Likely makes sense to have this as another question/discussion rather than tacking it on the original one ;)

Isn't this caused by some firewall as well?

**Reply by Abhishek627** [reply]
Sorry just an update. Issue was not with firewall but with salt. Instead of updating minion_id to match hostname, i removed /etc/salt/ folder and that caused issues with salt certs i guess.

**Reply by NicolasT** [reply]
Seems likely indeed.
